### ISSUE TITLE
Function to automatically hide posts from feed if heavily downvoted/has many reports

### ISSUE THREAD

**Issue by reallytiredofclowns** [discuitnet/discuit#139]
So there was a spam attack today and there unfortunately wasn't admin coverage to handle it.

Allowing for mass downvotes/reports to hide items from the feed empowers the regular userbase to help hide harmful content from the feed. Unfortunately, this also allows bad actors to mass downvote legitimate posts to hide it. It also creates a moderation/admin burden--they would have to check that hidden items are legitimately bad.

Perhaps creating a secondary class of trusted user, and only trusted users can tag posts for hiding? There would probably still have to be moderator/admin oversight, but the burden would be less, I think.

**Comment by CorrieTheCrabMan** [comment]
One of the issues we saw today was people couldn't hide the posts from the user because they quickly deleted their account, but the posts remained. Perhaps it should automatically hide posts with majority downvotes from deleted users? Wouldn't fix all the times this happens but might help

**Comment by reallytiredofclowns** [comment]
Yes, I feel that is a good alternative. If someone has deleted their account, then they no longer care about their content. So it's not even really open to abuse if people are "illegitimately" downvoting Ghost users' posts.

**Comment by previnder** [comment]
I think it's a good idea. Although, to prevent abuse of such a system, we should only take downvotes from old accounts beyond a certain threshold of points into account. 

For quick implementation, I would suggest the following:
1. Each post should have a `flagged` field of type `int` with a default value of 0.
2. There should be an option in the three-dots drop-down menu of a post to flag a post. This option should only be available to users whose accounts are at least 3 months old *and* have accumulated at least a 1000 points.
3. If the `flagged` count of a post is beyond a threshold, say beyond 5, then the post should be automatically hidden from all feeds. But I don't think it should go away completely; rather in its place should be a "this post is flagged" post with the real content of it hidden.

**Comment by reallytiredofclowns** [comment]
Yes, flagging and collapsing is an excellent implementation.

**Comment by reallytiredofclowns** [comment]
[Additional comments from asyoucanseE_ on the site](https://discuit.net/DiscuitMeta/post/5WDdCvg5/180b32d61032d72db0cd711c):

> The 1000 point threshold would be a significant change, because the current state is that points have no effect whatsoever, it's a decorative feature. I think it's okay as a temporary solution

> I like the idea of "just collapsing" the flagged posts, like how comments work in case of Muted users. (Also I wouldn't be against implementing the post collapse in case of muted users and discs, instead of not showing anything at all (or to have an option in the Settings to Hide or Collapse in case of muted user post, in case of muted disc post, in case of flagged post). But it's not a top priority for me, it just sounds like you can use the same solution, so it has to be easy, right? (like easy things exist in software development))

> (Edit: We could use this solution to collapse NSFW posts, user could choose the NSFW button on the Posting page. In case of image posts you can work around that currently, becuase we have galleries, but in case of text posts and link posts (where you can't choose the thumbnail), we can't)

> I agree the system can be abused, I'd suggest an alternative threshold: "user's account is at least 3 months old" AND ("user has at least 12 posts" OR "user has at least 48 comments"). It means user posts once a week or comments 4 times a week. If users "survive" 3 months and 12 posts (or 48 comments) without getting banned, they are probably a legit user.

> We could create a tiered system, where tier 1 (first few tiers) can be obtained automatically, and further tiers are given manually, based on criteria similar to how admins are chosen - it would be a state between admins and regular users, where they have bigger say in which posts should be flagged, but they can't delete, can't ban, can't use the "nuke user" button etc. The only difference between tiers is (let's say) "flagged += 5" or "flagged +=10" instead of "flagged +=1"

> My other concern about flagging if the flagged status of a post can't be reverted. So users should be able to vote "unflag", which is obv more useful for posts that are already flagged (flagged -= 1).

**Comment by previnder** [comment]
Maybe the 1000 points requirement is too much, and maybe we should cut it in half to 500 points. But that number has to be sufficiently high to prevent abuse. The 12 posts or 48 comments requirement would be trivial to get around; the 1000 or 500 points requirement would be much harder. 

This is a feature where there's no harm done if any new users or complete lurkers don't get to use it. The harm would be if bad faith actors get to utilize it against other users. The bias should disproportionately be on the side of restricting the feature too much as opposed to too little.

We should clearly state in the UI that the flagging system is only a measure against the most obvious trolls and spammers. Those posts must be removed by mods or admins in any case.
